src/Form/AdminSettingsForm.php

```diff
@@ -77,30 +74,43 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#empty' => $this->t('There are no items yet. Add roles.'),
       '#tabledrag' => [
         [
+          'table_id' => 'edit-role-theme-switcher',
           'action' => 'order',
           'relationship' => 'sibling',
           'group' => 'role-order-weight',
         ],
       ],
     ];
 
-    // Generates a form element for each role.
-    $roles = $this->getUserRoles();
-    foreach ($roles as $rid => $role) {
+    $this->setRowPerRole($this->getUserRoles(), $form);
```

Comment:
I don't agree to move this part in an other function. The code is not complicated enough to require this, and this does not bring a better code readability.